cmd/kubeadm/app/apis/kubeadm/validation/validation.go

```diff
@@ -267,3 +269,14 @@ func ValidateMixedArguments(flag *pflag.FlagSet) error {
 	}
 	return nil
 }
+
+func ValidateAPIEndpoint(c *kubeadm.API, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	endpoint := util.GenerateMasterEndpoint(c.AdvertiseAddress, c.BindPort)
```

Comment:
See comment below, but if we just check that the c.AdvertiseAddress is a valid IP, then we'd know that the endpoint would be valid - could be converted to a URL (and wouldn't have to actually generate the endpoint to validate).